src/java.base/share/classes/java/util/stream/AbstractPipeline.java

```diff
@@ -433,6 +433,21 @@ protected final boolean hasAnyStateful() {
          return result;
      }
 
+    /**
+     * Returns whether any of the stages in the (entire) pipeline is short-circuiting
+     * or not.
+     * @return {@code true} if any stage in this pipeline is stateful,
+     *         {@code false} if not.
+     */
+    protected final boolean isShortCircuitingPipeline() {
+        var result = false;
+        for (var u = sourceStage.nextStage;
+             u != null && !(result = StreamOpFlag.SHORT_CIRCUIT.isKnown(u.combinedFlags));
+             u = u.nextStage) {
+        }
+        return result;
```

Comment:
can be written in a simpler way
```
for(var stage = sourceStage.nextStage; stage != null; stage = stage.nextStage) {
  if (StreamOpFlag.SHORT_CIRCUIT.isKnown(u.combinedFlags)) {
    return true;
  }
  return false;
}
```
no local variable and no SSA phi

Reply:
In general, I don't like multiple exit-paths from methods, but I agree that in this case it increases readability.